server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/cassandra/CassandraBlobExistenceModule.java

```diff
@@ -0,0 +1,40 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.objectstorage.cassandra;
+
+import static com.datastax.driver.core.DataType.text;
+
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.utils.CassandraConstants;
+import org.apache.james.blob.objectstorage.cassandra.table.CassandraBlobExistenceTesterTable;
+
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+
+public interface CassandraBlobExistenceModule {
+    CassandraModule MODULE = CassandraModule.table(CassandraBlobExistenceTesterTable.TABLE_NAME)
+        .comment("Holds blob ids with their corresponding bucket to keep track of blobs existence within the object storage.")
+        .options(options -> options
+            .caching(SchemaBuilder.KeyCaching.ALL,
+                SchemaBuilder.rows(CassandraConstants.DEFAULT_CACHED_ROW_PER_PARTITION)))
+        .statement(statement -> statement
+            .addPartitionKey(CassandraBlobExistenceTesterTable.BUCKET_NAME, text())
+            .addClusteringColumn(CassandraBlobExistenceTesterTable.BLOB_ID, text()))
```

Comment:
Should be partition key to avoid wide row. Full scan on deleteBucket is acceptable.

Reply:
@chibenwa : could you explain why full scan would be acceptable?

Reply:
1. Because we don't know doing better

We expect buckets to have a high cardinality (at least 2* number of mails on the server). So a table model being:

```
PK bucketName
CK blobId
```
 - Would allow running the deleteBucket operation efficiently as a bucket entries will be co-located on the same node.

 - Is not acceptable as it will result in **wide rows** (and too much pressure on a signle node).

That's why people from the Cassandra world also have tools like Spark ;-)

2. Calling context.

Deleting a bucket is an administrative task triggered asynchronously via webadmin and is executed via the taskManager.

Being a long operation is expected.